scripts/ioctool

```diff
@@ -26,30 +46,81 @@ function iocdir {
 
 function ioccfg {
    iocdir "$1"
-   iocfile="${iocpath}"/"$1".cfg;
-   if [ ! -f "$iocfile" ]; then
-      iocfile=${iocpath}/iocBoot/${1}/st.cmd
-      if [ ! -f "$iocfile" ]; then
-         echo -e "Neither file found:\n${iocpath}/${1}.cfg\n$iocfile" >&2
+   iocfile=$(find ${iocpath} -name "$1.cfg")
+   if [ -z  "$iocfile" ]; then
+      iocfile=$(find ${iocpath} -name "st.cmd")
+      if [ -z "$iocfile" ]; then
+         echo -e "No files found for that ioc" >&2
+
          exit 1
       fi
    fi
 
+}
+
+function camname_to_pv {
+   #A function that will loop through camviewer.cfg files throughout all hutches to find the name then use that to search for its PV 
+   campv=""
+   hutch=`get_info --gethutch`
+   VIEWERCFG=/reg/g/pcds/pyps/config/$hutch/camviewer.cfg
+   CAMNAME=`echo $1 | sed -e 's/-/ /g' -e 's/_/ /g' -e 's/ /[-_ ]/g'`
+   
+   tmp=/tmp/cv.$$
+   cat $VIEWERCFG | grep -i "$CAMNAME" | grep -v '#' | awk -F, '{ if (split($2,a,";")==1) n=$2; else n=a[2]; gsub(" ","",n); gsub(" ","",$4); printf "%s %s\n", n, $4; }' >$tmp
```

Comment:
What does this line do and why does it need a temp file?

My initial reaction to stuff like this is that there has to be a more readable way to implement it. The triple pipe with the giant awk into a temp file doesn't seem particularly easy to implement, debug, or extend.

I think at minimum this needs a beefy comment to help the next person who needs to edit the script here.

Reply:
I put a beefy comment.